README.md

```diff
@@ -1,11 +1,78 @@
 # Dune Sync V2
 
+## Usage
 
+### Create a configuration file
 
+Configuration is provided in a single YAML file. Refer to the existing `config.yaml` for an overview.
+In general, it should contain a top-level key called `jobs`, whose members should be YAML dictionaries describing the
+job to run.
+
+#### Job parameters
+
+- `name`: An optional human-readable name for the job. Note that uniqueness is not enforced.
+- `source`: Definition of a data store to fetch data from. Exact parameter depends on source type, see **Sources**
+  further in this section.
+- `destination`: Definition of a data store to store data into. Exact parameter depends on source type, see *
+  *Destinations** further in this section.
+
+#### Sources
+
+Currently supported sources are Dune and PostgreSQL. Description of required parameters follows.
+
+#### Dune
+
+- `ref`: String. Must be exactly `dune`
+- `query_id`: Integer. ID of an existing Dune Query to execute and fetch results from.
+- `query_engine`: String and one of exactly `medium` or `large`.
+- `poll_frequency`: Integer. Determines how often, in seconds, to poll Dune for results after running the given query ID
+- `parameters`: Optional Dune Query parameters.
+    - `name`: String. Name of parameter
+    - `type`: String. Type of parameter. Must be one of: "text", "number", "datetime", "enum"
+    - `value`: Any. Value to send for the defined parameter
+
+#### Postgres
+
+- `ref`: String. Must be exactly `postgres`
+- `query_string`: String. Query to run against a Postgres server in order to fetch results and submit them to a
```

Comment:
We might want to make it possible to run a query file. I suspect some queries might be very long strings. In this case, the user would have to mount the config file along with, say a directory "queries" containing a bunch of raw SQL.

Reply:
Query file could be supported but we might also suggest using Yaml Heredoc syntax for that, like this:
```yaml
  - name: Write Results of Job 1 back
    source:
      ref: postgres
      if_exists: "append"
      query_string: |
        this
        will
        be
        some
        very
        long
        query

    destination:
      ref: dune
      table_name: "dune_sync_test_table"
```